### PR TITLE
feat: allow reverting to an earlier version of an app

### PR DESCRIPTION
**Pull request by fontanierh**
Screenshot 2023-04-21 at 15 43 20

Screenshot 2023-04-21 at 15 43 25

### CODE REVIEW

File: front/pages/api/w/[wId]/apps/[aId]/state.ts [hunk 1]
```diff
@@ -64,10 +64,25 @@ async function handler(
...
+        savedRun?: string;
+      } = {
         savedSpecification: req.body.specification,
         savedConfig: req.body.config,
-      });
+      };
+
+      if (req.body.run) {
+        if (typeof req.body.run != "string") {
+          res.status(400).end();
```
Comment (spolu) [inline front/pages/api/w/[wId]/apps/[aId]/state.ts:78]: Let's move to the new `apiError` for the entire file :grimacing:

**Review by spolu — DISMISSED** [review]
LGTM but would be nice to move to `apiError` for the `pages/api` files touched :+1:

File: front/pages/api/w/[wId]/apps/[aId]/state.ts (hunk 1 above)
Comment (fontanierh) [reply front/pages/api/w/[wId]/apps/[aId]/state.ts]: Done !

**Review by spolu — APPROVED** [review]
LGTM \o/ :fire:

File: front/pages/api/w/[wId]/apps/[aId]/state.ts [hunk 2]
```diff
@@ -3,7 +3,7 @@ import { Op } from "sequelize";
 
 import { Authenticator, getSession } from "@app/lib/auth";
 import { App } from "@app/lib/models";
-import { withLogging } from "@app/logger/withlogging";
+import { apiError, withLogging } from "@app/logger/withlogging";
 import { AppType } from "@app/types/app";
 
 export type PostStateResponseBody = {
```
Comment (spolu) [inline front/pages/api/w/[wId]/apps/[aId]/state.ts:9]: @fontanierh nit for a small follow-up PR: add `ReturnedAPIErrorType` here. Due to the `apiError` call, it does not complain, maybe there's something to do there as well?

Comment (fontanierh) [reply front/pages/api/w/[wId]/apps/[aId]/state.ts:9]: Good point !

**Pull request merged**